README.md

```diff
@@ -26,14 +26,75 @@ npm start
 ```
 And then open [http://localhost:3000](http://localhost:3000).
 
-Contribute
+How to Contribute
 ----------
 
 - Issue Tracker: [github.com/resin-io/docs/issues](https://github.com/resin-io/docs/issues)
 - Source Code: [github.com/resin-io/docs](https://github.com/resin-io/docs)
 
 If you think something is not documented, or can be improved, don't hesitate in sending a PR!
 
+### Get Setup
+
+```sh
+git clone https://github.com/resin-io/docs.git
+cd docs
+npm install
+```
+Now to check that everything it working correctly, start the local server with:
+```
+npm start
+```
+You should now see the docs served from localhost:3000
+
+To add you changes to the docs, create a new local branch of master:
+```
+git checkout -b my-docs-patch
+```
+Now you can make changes in this branch. Since our docs are static pages generated from markdown using metalsmith.io,  each time you change some of the `.md` files you will need to run rebuild of the docs to generate the static pages. This is done by running:
+```
+./tools/prepare.sh
+```
+However there is a handy watch functionality that will watch for changes in `/pages` and rebuild each time you save your changes. To use this, open a new tab in you terminal and run:
+```
+npm run watch-pages
+```
+Once you are happy with your changes, submit a pull request for you branch against `master`.
+
+### Using Partials
+
+When creating new content or altering current pages, its recommended that you try keep thing D.R.Y (Don’t Repeat Yourself). This is made easy by using “partials”. If you look in the `/shared` folder you will see a number of folders, these keep snippets or partials of the docs that can be reused all over the place. They also allow you to cleverly override a partial for a particular board type and language type. So in general you will see most of them just contain a `_default.md` which means that this partial will be used for all board and language types, but for partials like `getDeviceOnDash` we have device specific overrides.
+
+```
+├── getDeviceOnDash
+│   ├── _default.md
+│   ├── artik10.md
+│   ├── artik5.md
+│   ├── raspberrypi.md
+│   ├── raspberrypi2.md
+│   └── raspberrypi3.md
+```
+
+To use these partials in a markdown page all you need to do is add the following in the page:
+```
+{{ import “getDeviceOnDash” }}
+```
+and the engine will pull that partial into this location in the page.
+
+### Some other important bits
+
+Its important to note that `h1` and `h2` headings will automatically be included in the left hand side navigation bar, so make sure to use these appropriately.
+
+In the dynamically generated pages ( the ones with he device-type and language dropdown) it is possible to use templates to dynamically change the device name and language name in your text. Here is an example:
+```
+Getting started with {{ $device_details.name }} using {{ $language_details.name }}
```

Comment:
`$device` is the device ID (the key from the `dictionaries` file). Then there's `$device_details` (the name is auto-built) and `$device_dictionary` (all possible devices). I think it's better to be explicit here and the original `$device` property can be useful (especially in helpers and in other parts of the code).

But I can also refactor it so that `$device` becomes `$device_id` and `$device_details` is `$device`.